src/FlatSharp.Runtime/IFacadeTypeConverter.cs

```diff
@@ -0,0 +1,40 @@
+﻿/*
+ * Copyright 2021 James Courtney
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#nullable enable
+
+namespace FlatSharp.Runtime
+{
+    /// <summary>
+    /// Converts back and forth between a Facade Type (<typeparamref name="TFacadeType"/>) and an
+    /// underlying type used for storage (<typeparamref name="TUnderlyingType"/>).
+    /// </summary>
+    /// <typeparam name="TUnderlyingType">The underlying type used for literal storage in the FlatBuffer.</typeparam>
+    /// <typeparam name="TFacadeType">The facade type.</typeparam>
+    public interface IFacadeTypeConverter<TUnderlyingType, TFacadeType>
```

Comment:
As I mentioned in #86; maybe simply call it a `TypeFacade`? I think it plays well.

Reply:
Let me think about this one! I see what you mean, but I want it to be clear that's the interface itself isn't the facade -- just the back-and-forth.

Like they say, there are only two hard problems in computer science: cache consistency and naming things!